include/bm/bm_sim/actions.h

```diff
@@ -543,7 +547,7 @@ class ActionPrimitive_ {
   // (https://gcc.gnu.org/bugzilla/show_bug.cgi?id=60056).
   static thread_local Packet *pkt;
   static thread_local PHV *phv;
-
+  SourceInfo *srcInfo = nullptr;
```

Comment:
this should be renamed as it is misleading in its current form. This is not the source info for the action primitive, it is the source info for the action primitive *call*. The variable naming scheme follows the wrong convention as well. So maybe something like `call_source_info`?